contrib/docker/rootfs_prefix/rootfs_prefix.c

```diff
@@ -11,28 +11,22 @@
 #define BUFSIZE 256
 
 const char *add_prefix(const char *orig, char *prefixed) {
-  int status;
-  int errno;
-
-  if ((strncmp(orig, "/proc", 5) == 0) || (strncmp(orig, "/sys", 4) == 0)) {
-
-    status = snprintf(prefixed, BUFSIZE, "%s%s", PREFIX, orig);
-    if ((unsigned int)status >= BUFSIZE) {
-      error(status, ENAMETOOLONG,
-            "'%s' got truncated when adding '%s' prefix: '%s'", orig, PREFIX,
-            prefixed);
-      return orig;
-    } else if (status < 1) {
-      error(status, errno,
-            "adding '%s' prefix to file path failed: '%s' -> '%s'", PREFIX,
-            orig, prefixed);
-      return orig;
-    } else {
-      return (const char *)prefixed;
-    }
+  if ((strncmp(orig, "/proc", 5) != 0) && (strncmp(orig, "/sys", 4) != 0))
+    return orig;
 
-  } else {
+  int status = snprintf(prefixed, BUFSIZE, "%s%s", PREFIX, orig);
+  if ((unsigned int)status >= BUFSIZE) {
```

Comment:
You should do the `status < 1` check first: status may be negative due to error. When cast to an *unsigned int*, the resulting value is likely larger than `BUFSIZE`, resulting in an incorrect error message.